ninja/signature/details.py

```diff
@@ -222,9 +223,24 @@ def _get_param_type(self, name: str, arg: inspect.Parameter) -> FuncParam:
 
         if get_origin(annotation) is Annotated:
             args = get_args(annotation)
-            if isinstance(args[1], Param):
+            if isinstance(args[-1], Param):
                 prev_default = default
-                annotation, default = args
+                if len(args) == 2:
+                    annotation, default = args
+                else:
+                    # TODO: Remove version check once support for <=3.8 is dropped.
+                    # Annotated[] is only available at runtime in 3.9+ per
+                    # https://docs.python.org/3/library/typing.html#typing.Annotated
+                    if version_info >= (3, 9):
+                        # NOTE: Annotated[args[:-1]] seems to have the same runtime
+                        # behavior as Annotated[*args[:-1]], but the latter is
+                        # invalid in Python < 3.11 because star expressions
+                        # were not allowed in index expressions.
+                        annotation, default = Annotated[args[:-1]], args[-1]
+                    else:  # pragma: no cover -- requires specific Python versions
+                        raise NotImplementedError(
+                            "This definition requires Python version 3.9+"
+                        )
```

Comment:
I added tests that cover this branch, but the tests are only defined for specific Python versions, hence the need for `# pragma: no cover`.

I'm not 100% sure `NotImplementedError` is the appropriate exception here. See https://docs.python.org/3/library/exceptions.html#NotImplementedError. Let me know if there's a different exception you'd prefer.

Reply:
fine with - looking at the stats 3.8 is less then 1% in downloads - chances that some one using this are minimal (and maybe will push to upgrade python version)